File: cifrado/Escritorio/C++/CifradoCesar/CifradoCesar/Cifrador.cpp
#include "stdafx.h"
#include "Cifrador.h"

#include <string>
#include <string.h>

#include "Alfabeto.h"
#include "TextoPlano.h"
using namespace std;

Cifrador::Cifrador(){}

string Cifrador::CifradoTipoCesar(string texto, int desplazamiento, string abecedario)
{
	TextoPlano textoPlano(texto);
	Alfabeto alfabeto(abecedario);


	string textoCifrado = "";
	int posicion = 0;

	for (int i = 0; i < texto.length(); i++)
	{
		if (alfabeto.perteneceAlfabeto(texto[i]))
		{
			posicion = alfabeto.buscarPosicionLetra(texto[i]);
			posicion = posicion + desplazamiento;

			if (posicion >abecedario.length())
				posicion = posicion - abecedario.length();

			textoCifrado = textoCifrado + abecedario[posicion];
		}
		else
			textoCifrado = textoCifrado + texto[i];
	}

	return textoCifrado;
}

string Cifrador::cifradoTipoTransposicionInversa(string texto)
{
	TextoPlano textoPlano(texto);
	return textoPlano.invertirTexto();
}

string Cifrador::cifradoTipoTransposicionInversaPorGrupos(string texto, int desplazamiento)
{
	TextoPlano textoPlano(texto);
	return textoPlano.invertirTextoGrupos(desplazamiento);
}

File: cifrado/Escritorio/codigo/C#/CifradoCesarCshare/CifradoCesarCshare/Cifrador.cs


namespace CifradoCesarCshare
{
    class Cifrador
    {
        Alfabeto alfabeto;
        TextoPlano textoPlano;

        public Cifrador(Alfabeto abecedario)
        {
            alfabeto = abecedario;
        }

        public string CifradoTipoCesar(string texto, int desplazamiento)
        {
            textoPlano = new TextoPlano(texto);
            string textoCifrado = "";
            int posicion = 0;

            for (int i = 0; i < textoPlano.textoPlano.Length; i++)
            {
                if (alfabeto.perteneceAlfabeto(textoPlano.textoPlano[i]))
                {
                    posicion = alfabeto.buscarPosicionLetra(textoPlano.textoPlano[i]);
                    posicion = posicion + desplazamiento;

                    if (posicion > alfabeto.caracteres.Length)
                        posicion = posicion - alfabeto.caracteres.Length;

                    textoCifrado = textoCifrado + alfabeto.caracteres[posicion];
                }
                else
                    textoCifrado = textoCifrado + textoPlano.textoPlano[i];
            }

            return textoCifrado;
        }

        public string cifradoTipoTransposicionInversa(string texto)
        {
            textoPlano = new TextoPlano(texto);
            return textoPlano.invertirTexto();
        }


        public string cifradoTipoTransposicionInversaPorGrupos(string texto,int desplazamiento)
        {
            textoPlano = new TextoPlano(texto);
            return textoPlano.invertirTextoGrupos(desplazamiento);
        }

       
        
    }
}
